### PR TITLE
ROX-12612: Add collections datastore layer with stubs and todo comments

### PR DESCRIPTION
**Pull request by keyallis**
## Description

Create Collections datastore structure with some TODOs for upcoming tasks.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added

### CODE REVIEW

File: central/resourcecollection/datastore/datastore.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"github.com/stackrox/rox/central/resourcecollection/datastore/search"
+	"github.com/stackrox/rox/central/resourcecollection/datastore/store/postgres"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	searchPkg "github.com/stackrox/rox/pkg/search"
+)
+
+//go:generate mockgen-wrapper
+
+// DataStore is a wrapper around a store that provides search functionality
```
Comment (theencee) [inline central/resourcecollection/datastore/datastore.go:19]: it provides more than search

**Review by theencee — APPROVED** [review]
Rebase to pick up fixes for the test failures

File: central/resourcecollection/datastore/index/index.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import (
+	"context"
+
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/search/blevesearch"
+)
+
+// Indexer provides functionality to index node components.
```
Comment (charmik-redhat) [inline central/resourcecollection/datastore/index/index.go:12]: minor : to index resource collections

File: central/resourcecollection/datastore/singleton.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	ds DataStore
+)
+
+func initialize() {
+	storage := postgres.New(globaldb.GetPostgres())
+	indexer := postgres.NewIndexer(globaldb.GetPostgres())
+	ds = New(storage, indexer, search.New(storage, indexer))
+}
+
+// Singleton returns a singleton instance of cve datastore
```
Comment (charmik-redhat) [inline central/resourcecollection/datastore/singleton.go:22]: minor : instance of resource collection datastore

**Review by charmik-redhat — COMMENTED** [review]
Minor comments, can be addressed along with another PR

**Pull request merged**